src/main/scala/com/comcast/xfinity/sirius/uberstore/segmented/SegmentedCompactor.scala

```diff
@@ -165,6 +206,7 @@ private [segmented] class SegmentedCompactor(maxDeleteAgeMillis: Long) {
     left.foreach(target.writeEntry)
     right.foreach(target.writeEntry)
     target.setApplied(applied = left.isApplied && right.isApplied)
+    target.setInternallyCompacted(compacted = false)
```

Comment:
Setting this to 'false' may be a bit conservative.  I believe that if the left and right are both Segment compacted, and also are both Internally compacted, then we could safely set the InternallyCompacted flag to true here.

After the initial internal compaction is done across all Segments and the old Deletes are removed, there is quite a bit of merging that happens in the repository.  That leads to quite a bit of unnecessary Segment compaction activity.  It only happens once but it might be better to avoid it.

I believe the statement could be changed to this:
target.setInternallyCompacted(compacted = left.isApplied && right.isApplied && left.isInternallyCompacted && right.isInternallyCompacted)